includes/admin/helper/class-wc-helper.php

```diff
@@ -1237,6 +1237,27 @@ public static function admin_notices() {
 	 * Prompt a Helper connection if the user has WooCommerce.com extensions.
 	 */
 	private static function _prompt_helper_connect( $screen_id ) {
+		$screens = array(
+			'plugins',
+			'product',
+			'edit-product',
+			'edit-product_cat',
+			'edit-product_tag',
+			'product_page_product_attributes',
+			'shop_order',
+			'edit-shop_order',
+			'shop_coupon',
+			'edit-shop_coupon',
+			'woocommerce_page_wc-reports',
+			'woocommerce_page_wc-settings',
+			'woocommerce_page_wc-status',
+			'woocommerce_page_wc-addons',
+		);
+
+		if ( ! in_array( $screen_id, $screens ) ) {
+			return;
+		}
+
 		// Don't show the notice on the Helper screens.
 		if ( 'woocommerce_page_wc-addons' == $screen_id && ! empty( $_REQUEST['section'] ) && 'helper' == $_REQUEST['section'] ) {
```

Comment:
i know this isn't included in the PR, but a quick note: you may also want to make adjustments to use `sanitize_title( __( 'WooCommerce', 'woocommerce' ) ) . '_page_wc-settings'` here instead to compare against screen ID like other parts of WC do (for sites with "WooCommerce" translated, esp when languages use non-ASCII characters). See https://core.trac.wordpress.org/ticket/21454 and https://core.trac.wordpress.org/ticket/18857 (and maybe one day @mikejolley's patch can finally be applied :wink:)